src/fitnesse/slim/fixtureInteraction/SimpleInteraction.java

```diff
@@ -261,17 +266,39 @@ private Method findMatchingMethod(String methodName, Method[] methods, int nArgs
     return method;
   }
 
+
   private boolean isMatchingMethod(Method method, String methodName, int nArgs) {
     boolean hasMatchingName = method.getName().equals(methodName);
     boolean hasMatchingArguments = method.getParameterTypes().length == nArgs;
     return hasMatchingName && hasMatchingArguments;
   }
 
   protected MethodExecutionResult invokeMethod(Object instance, Method method, Object[] args) throws Throwable {
-    Object[] convertedArgs = convertArgs(method, args);
-    Object retval = callMethod(instance, method, convertedArgs);
-    Class<?> retType = method.getReturnType();
-    return new MethodExecutionResult(retval, retType);
+      Object[] convertedArgs = null;
+      Object retval = null;
+      String methodName = method.getDeclaringClass().getName() + "." + MethodExecutionResult.methodToString(method)
+                          + "." + ((instance == null) ? "" : " On instance of: " + instance.getClass().getName());
+      Class<?> retType = method.getReturnType();
+      try {
+        convertedArgs = convertArgs(method, args);
+      } catch (SlimError e) {
+	throw e;
+	// TODO: create test cases for this to check that it works and is displayed correctly in the result
+	//throw new SlimException(e.getMessage()+ "\n\n"+ "You may want to check if the correct method signature has been identified by Slim:\n" + methodName + "\n", e, true);
+      } catch (Exception e) {
+	throw e;
+	// TODO: create test cases for this to check that it works and is displayed correctly in the result 
+        //throw new SlimError(e.getMessage()+ "\n\n"+ "You may want to check if the correct method signature has been identified by Slim:\n" + methodName + "\n", e);
+      }
+      try {
+        retval = callMethod(instance, method, convertedArgs);
+      } catch (Exception e) {
+        // We can't modify the exception text as it could be an exception coming from the SUT
+	// TODO print this only if there are at least to methods with the same name and number of parameters  
+        System.out.println("InvokeMethod failed during invocation, you may want to check if the correct method signature has been identified by Slim: "  + " : " + methodName);
```

Comment:
Instead of writing to System.out it seems more appropriate to create a logger (e.g. `  private static final Logger LOG = Logger.getLogger(SimpleInteraction.class.getName());
`) and use `LOG.warn(...`.

Reply:
Hi @fhoeben, could you run travis-ci. I would like to see if the fix to the test case I made worked.
Your other open comments I will rework or remove depending on my time in the next week.

Reply:
the SlimService never uses a Logger. I assume to not interfere with the system under test.
Nevertheless you are right I will remove this or make it only appear in verbose mode.